app/connectors/ApiDefinitionConnector.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.WSHttp
+import model._
+import uk.gov.hmrc.play.http._
+import uk.gov.hmrc.play.config.ServicesConfig
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait ApiDefinitionConnector {
+  val serviceBaseUrl: String
+  val http: HttpGet
+
+  def fetchAll()(implicit hc: HeaderCarrier): Future[Seq[APIDefinition]] = {
+    http.GET[Seq[APIDefinition]](s"$serviceBaseUrl/api-definition")
+      .recover {
+        case e: Upstream5xxResponse => throw new FetchApplicationsFailed
```

Comment:
Yes, you're right... have altered and added some tests + fixed the failing acceptance tests